src/playercards/DeckBuilder.ttslua

```diff
@@ -0,0 +1,682 @@
+    local cardMetadata = allCardsBagApi.getCardById(cardId).metadata
+    if cardMetadata.type == "Skill" then
+      table.insert(cardList.skillList, cardId)
+    elseif cardMetadata.type == "Event" then
+      table.insert(cardList.eventList, cardId)
+    elseif cardMetadata.type == "Asset" then
+      table.insert(cardList.assetList, cardId)
+    else
+      table.insert(cardList.otherList, cardId)
+    end
+  end
+
+  --Organizes the card lists, and presents them to the player
+  cardList.skillList = allCardsBagApi.sortCardsTable(cardList.skillList)
+  cardList.eventList = allCardsBagApi.sortCardsTable(cardList.eventList)
+  cardList.assetList = allCardsBagApi.sortCardsTable(cardList.assetList)
+  cardList.otherList = allCardsBagApi.sortCardsTable(cardList.otherList)
+  playerCardPanelApi.spawnCardsByType({
+    cardList = cardList,
+    name = "deckOptions",
+    startPos = "deckBuilder",
+    spread = true,
+    spreadCols = 35,
+    targetState = "middle"
+  })
+
+end
+
+function createFilterString(deckOptions, choiceSelection)
+  local filterString = "("
+  local notString = "("
+  local specList = { }
+  for i, cardSpec in ipairs(deckOptions) do
+    if cardSpec.choiceName == choiceSelection[cardSpec.choiceName] then
+      local filterSubString = " ("
+
+      if cardSpec.faction then
+        filterSubString = filterSubString .. processCardSpec(cardSpec.faction, "f", filterSubString == " (")
+      end
+
+      if cardSpec.trait then
+        filterSubString = filterSubString .. processCardSpec(cardSpec.trait, "k", filterSubString == " (")
+      end
+
+      if cardSpec.uses then
+        filterSubString = filterSubString .. processCardSpec(cardSpec.uses, "u", filterSubString == " (")
+      end
+
+      if cardSpec.special then
+        filterSubString = filterSubString .. processCardSpec(cardSpec.special, "q", filterSubString == " (")
+      end
+
+      if cardSpec.permanent then
+        filterSubString = filterSubString .. processCardSpec(cardSpec.special, "v", filterSubString == " (")
+      end
+
+      if cardSpec.type then
+        filterSubString = filterSubString .. processCardSpec(cardSpec.type, "t", filterSubString == " (")
+      end
+
+      if cardSpec.level then
+        filterSubString = filterSubString .. " & ( p>" .. cardSpec.level.min - 1 .. " & p<" .. cardSpec.level.max + 1 .. " )"
+      end
+
+      filterSubString = filterSubString .. " )"
+      
+      if cardSpec["not"] then
+        notString = notString .. filterSubString .. " |"
+      else
+        filterString = filterString .. filterSubString .. " |"
+      end
+    end
+  end
+  filterString = filterString .. " )"
+  notString = notString .. " )"
+  return {
+    baseString = filterString,
+    exclusionString = notString
+  }
+end
+
+function processCardSpec(specList, filterType, isFirst)
+  local filterSubString = ""
+  if not isFirst then
+    if filterType == "t" then
+      filterSubString = " &"
+    else
+      filterSubString = " |"
```

Comment:
Why are there spaces on top of the delimiter? And why does "type" get a different one?